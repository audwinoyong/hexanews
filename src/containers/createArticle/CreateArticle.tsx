/**
 * Create Article containers for users to create article
 */

import * as React from 'react'
import Avatar from '@material-ui/core/Avatar'
import Button from '@material-ui/core/Button'
import TextField from '@material-ui/core/TextField'
import Typography from '@material-ui/core/Typography'
import InputLabel from '@material-ui/core/InputLabel'
import MenuItem from '@material-ui/core/MenuItem'
import FormHelperText from '@material-ui/core/FormHelperText'
import FormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'
import CircularProgress from '@material-ui/core/CircularProgress'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import { push } from 'connected-react-router'
const ReactQuill = require('react-quill')

import * as actions from './actions'
import styles from './styles'
import { createArticleString } from '../../constants/string'
import selector, { StateProps } from './selector'

export type Props = {
  dispatch: Dispatch<any>
} & StateProps

export interface ComponentState {
  category: string
  description: string
  title: string
}

export class CreateArticle extends React.Component<Props, ComponentState> {

  constructor(props: Props) {
    super(props)
    this.state = {
      category: '',
      description: '',
      title: '',
    }
  }

  handleCategoryChange = (event: any) => {
    this.setState({ category: event.target.value })
  }

  handleTitleChange = (event: any) => {
    this.setState({ title: event.target.value })
  }

  handleDescriptionChange = (html: any) => {
    this.setState({ description: html })
  }

  disablePublishButton() {
    const { category, description, title } = this.state
    return (category === '' || description === '' || title === '')
  }

  onCreateArticle = async () => {
    const { dispatch } = this.props
    const { title, category, description } = this.state
    const authorId = localStorage.getItem('id') || ''

    const newArticle = { title, category, description, authorId }
    await dispatch(actions.createArticle(newArticle))
  }

  modules = {
    toolbar: [
      [{ 'header': '1'}, {'header': '2'}, { 'font': [] }],
      [{size: []}],
      ['bold', 'italic', 'underline', 'strike', 'blockquote'],
      [{'list': 'ordered'}, {'list': 'bullet'},
      {'indent': '-1'}, {'indent': '+1'}],
      ['link', 'image', 'video'],
      ['clean']
    ],
    clipboard: {
      matchVisual: false,
    }
  }

  formats = [
    'header', 'font', 'size',
    'bold', 'italic', 'underline', 'strike', 'blockquote',
    'list', 'bullet', 'indent',
    'link', 'image', 'video'
  ]

  public render() {
    const { dispatch, user, isCreatingArticle } = this.props
    const { title, category, description } = this.state
    return (
      <div style={styles.container}>
        <div style={styles.profileContainer}>
          <Avatar style={styles.avatar}>HC</Avatar>
          <div style={styles.profile as any}>
            <Typography variant='subheading' style={styles.title}>
              {user.name}
            </Typography>
            <Typography variant='body1' color='textSecondary'>
              {user.description}
            </Typography>
            <Typography variant='body1' color='textSecondary'>
              {'Draft'}
            </Typography>
          </div>
        </div>
        <div>
          <form>
            <TextField
              id='title'
              label='Title'
              value={title}
              onChange={this.handleTitleChange}
              style={styles.textField}
              margin='normal'
            />
            <FormControl required style={styles.categoryTextField}>
              <InputLabel htmlFor='category-required'>Category</InputLabel>
              <Select
                value={category}
                onChange={this.handleCategoryChange}
                name='category'
              >
                <MenuItem value='technology'>Technology</MenuItem>
                <MenuItem value='science'>Science</MenuItem>
                <MenuItem value='art'>Art</MenuItem>
                <MenuItem value='design'>Design</MenuItem>
                <MenuItem value='culture'>Culture</MenuItem>
                <MenuItem value='photography'>Photography</MenuItem>
                <MenuItem value='leadership'>Leadership</MenuItem>
                <MenuItem value='math'>Math</MenuItem>
                <MenuItem value='economy'>Economy</MenuItem>
                <MenuItem value='music'>Music</MenuItem>
              </Select>
              <FormHelperText>Required</FormHelperText>
            </FormControl>
            <ReactQuill
              value={description}
              onChange={this.handleDescriptionChange}
              formats={this.formats}
              modules={this.modules}
              placeholder='Tell your story in here...'
              theme='snow'
              style={styles.descTextField}
            />
          </form>
        </div>
        <div style={styles.buttonContainer}>
          <Button
            variant='outlined'
            component='button'
            style={styles.button}
            disabled={this.disablePublishButton()}
            onClick={this.onCreateArticle}
          >
            {isCreatingArticle ? <CircularProgress size={22} /> : createArticleString.publishButton}
          </Button>
          <Button
            id='cancelButton'
            variant='outlined'
            component='button'
            style={styles.button}
            onClick={() => dispatch(push('/'))}
          >
            {createArticleString.cancelButton}
          </Button>
        </div>
      </div>
    )
  }
}

export default connect(selector)(CreateArticle)