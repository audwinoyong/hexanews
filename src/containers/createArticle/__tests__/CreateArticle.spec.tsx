/**
 * Unit tests for checking the behavior of CreateArticle container
 */

import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import { shallow } from 'enzyme'

import { CreateArticle, Props, ComponentState } from '../CreateArticle'

jest.mock('react-quill')
Enzyme.configure({ adapter: new Adapter() })

const mockUser = {
  _id: '1',
  createdAt: 'created',
  description: 'desc',
  email: 'user@email.com',
  name: 'user'
}

const mockCategory = {
  title: 'tech',
  description: 'test',
}

describe('Create Article', () => {
  it('renders Create Article', () => {
    const component = renderer.create(
      <CreateArticle
        user={mockUser}
        categories={[mockCategory]}
        dispatch={jest.fn()}
        isCreatingArticle={false}
        createArticleError=''
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders Create Article with isCreatingArticle as true', () => {
    const component = renderer.create(
      <CreateArticle
        user={mockUser}
        categories={[mockCategory]}
        dispatch={jest.fn()}
        isCreatingArticle={true}
        createArticleError=''
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handle onChangeTitle', () => {
    const component = (
      <CreateArticle
        user={mockUser}
        categories={[mockCategory]}
        dispatch={jest.fn()}
        isCreatingArticle={true}
        createArticleError=''
      />
    )

    const layout = shallow<Props, ComponentState>(component)
    const instance: CreateArticle = layout.instance() as CreateArticle

    const title = layout.state().title
    expect(title).toBe('')

    const event = {
      target: {
        value: 'asdf'
      }
    }

    instance.handleTitleChange(event)
    layout.update()

    const updatedTitle = layout.state().title
    expect(updatedTitle).toBe('asdf')
  })

  it('handle onChangeCategory', () => {
    const component = (
      <CreateArticle
        user={mockUser}
        categories={[mockCategory]}
        dispatch={jest.fn()}
        isCreatingArticle={true}
        createArticleError=''
      />
    )

    const layout = shallow<Props, ComponentState>(component)
    const instance: CreateArticle = layout.instance() as CreateArticle

    const category = layout.state().category
    expect(category).toBe('')

    const event = {
      target: {
        value: 'technology'
      }
    }

    instance.handleCategoryChange(event)
    layout.update()

    const updatedCategory = layout.state().category
    expect(updatedCategory).toBe('technology')
  })

  it('handle onChangeDescription', () => {
    const component = (
      <CreateArticle
        user={mockUser}
        categories={[mockCategory]}
        dispatch={jest.fn()}
        isCreatingArticle={true}
        createArticleError=''
      />
    )

    const layout = shallow<Props, ComponentState>(component)
    const instance: CreateArticle = layout.instance() as CreateArticle

    const description = layout.state().description
    expect(description).toBe('')

    const html = '<p>test</p>'

    instance.handleDescriptionChange(html)
    layout.update()

    const updatedDescription = layout.state().description
    expect(updatedDescription).toBe(html)
  })

  it('handle onCreateArticle action correctly', () => {
    const dispatch = jest.fn()
    const component = (
      <CreateArticle
        user={mockUser}
        categories={[mockCategory]}
        dispatch={dispatch}
        isCreatingArticle={true}
        createArticleError=''
      />
    )

    const layout = shallow(component)
    const instance: CreateArticle = layout.instance() as CreateArticle

    instance.onCreateArticle()
    layout.update()

    expect(dispatch).toHaveBeenCalled()
  })

  it('handle button cancel click', () => {
    const dispatch = jest.fn()
    const component = (
      <CreateArticle
        user={mockUser}
        categories={[mockCategory]}
        dispatch={dispatch}
        isCreatingArticle={true}
        createArticleError=''
      />
    )
    const layout = shallow(component)
    layout.find('#cancelButton').first().simulate('click')
    expect(dispatch).toHaveBeenCalled()
  })
})