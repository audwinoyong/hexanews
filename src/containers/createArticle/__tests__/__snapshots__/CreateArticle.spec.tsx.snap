// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Article renders Create Article 1`] = `
<div
  style={
    Object {
      "marginTop": "4%",
      "paddingBottom": "5%",
    }
  }
>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 MuiGrid-grid-md-6-63"
      style={
        Object {
          "display": "flex",
          "marginBottom": "3%",
        }
      }
    >
      <div
        className="MuiAvatar-root-98 MuiAvatar-colorDefault-99"
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "green",
            "height": 50,
            "margin": 10,
            "width": 50,
          }
        }
      >
        U
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <h3
          className="MuiTypography-root-101 MuiTypography-subheading-108"
          style={
            Object {
              "marginTop": 12,
            }
          }
        >
          user
        </h3>
        <p
          className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-colorTextSecondary-124"
        >
          desc
        </p>
        <p
          className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-colorTextSecondary-124"
        >
          Draft
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
  </div>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 MuiGrid-grid-md-6-63"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-126 MuiFormControl-marginNormal-127"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <label
          className="MuiFormLabel-root-135 MuiInputLabel-root-130 MuiInputLabel-formControl-131 MuiInputLabel-animated-134"
          data-shrink={false}
          htmlFor="title"
        >
          Title
        </label>
        <div
          className="MuiInput-root-142 MuiInput-formControl-143 MuiInput-underline-146"
        >
          <input
            aria-invalid={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-150"
            defaultValue={undefined}
            disabled={false}
            id="title"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={false}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-126"
        style={
          Object {
            "marginTop": 24,
            "width": "100%",
          }
        }
      >
        <label
          className="MuiFormLabel-root-135 MuiFormLabel-required-140 MuiInputLabel-root-130 MuiInputLabel-formControl-131 MuiInputLabel-animated-134"
          data-shrink={false}
          htmlFor="category-required"
        >
          Category
          <span
            className="MuiFormLabel-asterisk-141"
          >
            â€‰*
          </span>
        </label>
        <div
          className="MuiInput-root-142 MuiInput-formControl-143 MuiInput-underline-146"
        >
          <div
            className="MuiSelect-root-155"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-156 MuiSelect-selectMenu-157 MuiInput-input-150"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              autoComplete={undefined}
              autoFocus={undefined}
              defaultValue={undefined}
              id={undefined}
              name="category"
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              rows={undefined}
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-160 MuiSelect-icon-159"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <p
          className="MuiFormHelperText-root-171 MuiFormHelperText-required-177"
        >
          Required
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
  </div>
  <div
    className="MuiGrid-container-1"
    style={
      Object {
        "marginTop": "5%",
      }
    }
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 MuiGrid-grid-md-6-63"
    >
      <h2
        className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-colorTextPrimary-123"
        style={
          Object {
            "color": "#E53935",
          }
        }
      >
        
      </h2>
      <button
        className="MuiButtonBase-root-204 MuiButtonBase-disabled-205 MuiButton-root-178 MuiButton-text-180 MuiButton-outlined-186 MuiButton-disabled-198"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginRight": 10,
            "marginTop": 30,
          }
        }
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-179"
        >
          Publish
        </span>
      </button>
      <button
        className="MuiButtonBase-root-204 MuiButton-root-178 MuiButton-text-180 MuiButton-outlined-186"
        disabled={false}
        id="cancelButton"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginRight": 10,
            "marginTop": 30,
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-179"
        >
          Cancel
        </span>
        <span
          className="MuiTouchRipple-root-207"
        />
      </button>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
  </div>
</div>
`;

exports[`Create Article renders Create Article with isCreatingArticle as true 1`] = `
<div
  style={
    Object {
      "marginTop": "4%",
      "paddingBottom": "5%",
    }
  }
>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 MuiGrid-grid-md-6-63"
      style={
        Object {
          "display": "flex",
          "marginBottom": "3%",
        }
      }
    >
      <div
        className="MuiAvatar-root-98 MuiAvatar-colorDefault-99"
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "green",
            "height": 50,
            "margin": 10,
            "width": 50,
          }
        }
      >
        U
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <h3
          className="MuiTypography-root-101 MuiTypography-subheading-108"
          style={
            Object {
              "marginTop": 12,
            }
          }
        >
          user
        </h3>
        <p
          className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-colorTextSecondary-124"
        >
          desc
        </p>
        <p
          className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-colorTextSecondary-124"
        >
          Draft
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
  </div>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 MuiGrid-grid-md-6-63"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-126 MuiFormControl-marginNormal-127"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <label
          className="MuiFormLabel-root-135 MuiInputLabel-root-130 MuiInputLabel-formControl-131 MuiInputLabel-animated-134"
          data-shrink={false}
          htmlFor="title"
        >
          Title
        </label>
        <div
          className="MuiInput-root-142 MuiInput-formControl-143 MuiInput-underline-146"
        >
          <input
            aria-invalid={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-150"
            defaultValue={undefined}
            disabled={false}
            id="title"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={false}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-126"
        style={
          Object {
            "marginTop": 24,
            "width": "100%",
          }
        }
      >
        <label
          className="MuiFormLabel-root-135 MuiFormLabel-required-140 MuiInputLabel-root-130 MuiInputLabel-formControl-131 MuiInputLabel-animated-134"
          data-shrink={false}
          htmlFor="category-required"
        >
          Category
          <span
            className="MuiFormLabel-asterisk-141"
          >
            â€‰*
          </span>
        </label>
        <div
          className="MuiInput-root-142 MuiInput-formControl-143 MuiInput-underline-146"
        >
          <div
            className="MuiSelect-root-155"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-156 MuiSelect-selectMenu-157 MuiInput-input-150"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              autoComplete={undefined}
              autoFocus={undefined}
              defaultValue={undefined}
              id={undefined}
              name="category"
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              rows={undefined}
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-160 MuiSelect-icon-159"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <p
          className="MuiFormHelperText-root-171 MuiFormHelperText-required-177"
        >
          Required
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
  </div>
  <div
    className="MuiGrid-container-1"
    style={
      Object {
        "marginTop": "5%",
      }
    }
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-10-39 MuiGrid-grid-md-6-63"
    >
      <h2
        className="MuiTypography-root-101 MuiTypography-body1-110 MuiTypography-colorTextPrimary-123"
        style={
          Object {
            "color": "#E53935",
          }
        }
      >
        
      </h2>
      <button
        className="MuiButtonBase-root-204 MuiButtonBase-disabled-205 MuiButton-root-178 MuiButton-text-180 MuiButton-outlined-186 MuiButton-disabled-198"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginRight": 10,
            "marginTop": 30,
          }
        }
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-179"
        >
          <div
            className="MuiCircularProgress-root-221 MuiCircularProgress-colorPrimary-224 MuiCircularProgress-indeterminate-223"
            role="progressbar"
            style={
              Object {
                "height": 22,
                "width": 22,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg-226"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle-227 MuiCircularProgress-circleIndeterminate-229"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </span>
      </button>
      <button
        className="MuiButtonBase-root-204 MuiButton-root-178 MuiButton-text-180 MuiButton-outlined-186"
        disabled={false}
        id="cancelButton"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginRight": 10,
            "marginTop": 30,
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-179"
        >
          Cancel
        </span>
        <span
          className="MuiTouchRipple-root-207"
        />
      </button>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-1-30 MuiGrid-grid-md-3-60"
    />
  </div>
</div>
`;
